components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/SystemEngineView.kt

```diff
@@ -324,6 +325,34 @@ class SystemEngineView @JvmOverloads constructor(
             session?.internalNotifyObservers { onContentPermissionRequest(SystemPermissionRequest(request)) }
         }
 
+        override fun onJsAlert(view: WebView?, url: String?, message: String?, result: JsResult): Boolean {
+            session?.apply {
+                val hasShownManyDialogs = (++alertJSCount) > MAX_JS_ALERT
+                val onDismiss: () -> Unit = {
+                    result.cancel()
+                }
+
+                if (shouldShowNoMoreJSDialogs) {
+
+                    notifyObservers {
+                        onPromptRequest(
+                            PromptRequest.Alert(
+                                "",
+                                message ?: "",
+                                hasShownManyDialogs,
```

Comment:
subjective nit: maybe something like `showMultipleDialogs`, `tooManyDialogs`? 4+ words make it hard to grok for me at least. 😅